pkg/scte35/segmentation_descriptor.go

```diff
@@ -164,6 +164,19 @@ const (
 	SegmentationTypeNetworkStart = 0x50
 	// SegmentationTypeNetworkEnd is the segmentation_type_id for Network End.
 	SegmentationTypeNetworkEnd = 0x51
+
+	// The following values are not proper segmentation_type_id as used in the
+	// SegmentationsDescriptor. They use value that are higher than the
+	// segmentation_type_id range to avoid conflicts with the actual values.
+
+	// SegmentationTypeSpliceInsertOut is a pseudo segmentation_type_id used
+	// for reporting the Duration of a Splice Insert command that has
+	// Out of Network Indicator set.
+	SegmentationTypeSpliceInsertOut = 0x100
+	// SegmentationTypeSpliceInsertIn is a pseudo segmentation_type_id used
+	// for reporting the Duration of a Splice Insert command that has
+	// Out of Network Indicator cleared.
+	SegmentationTypeSpliceInsertIn = 0x101
```

Comment:
I don't love these. See my comment below on a more robust `Durations` struct.

If we do want to stick with a map, we should consider using names that makes it more obvious that these are Durations keys and *not* SegmentationTypes.

Reply:
I would argue that those actually are segmentation types in a sense. It's the exact same function as provided by time signal segmentation signaling, it's a start or end of a segment type with optional duration value.

Reply:
That was poorly phrased on my part. My issue is primarily with the `SegmentationType` prefix. It implies that these values are valid for use when constructing `SegmentationDescriptors` like so:

```
&scte35.SegmentationDescriptor{
  SegmentationTypeID: SegmentationTypeSpliceInsertOut
}
```

Reply:
As I pointed out in the other thread, the Segmentation Type ID in the struct is 32 bit, even though SCTE35 only has 8 bit for them. The encode currently just ignores the top 24 bit. So regardless of if we want to use special Types here or not, I guess there should be a check that the value is not too big. Such a check would also catch a misuse of these pseudo type IDs.